src/tracing/trace_event.h

```diff
@@ -467,6 +474,7 @@ static inline uint64_t AddTraceEventImpl(
   // DCHECK(num_args, 2);
   v8::TracingController* controller =
       node::tracing::TraceEventHelper::GetTracingController();
+  if (controller == nullptr) return 0;
```

Comment:
may be move this to the beginning of the function? here and in `AddMetadataEventImpl` ?

Reply:
@gireeshpunathil How would I do that without introducing a potential memory leak?

Reply:
@addaleax - I am sorry, but not following: we have a nullcheck and return (with no trace action) in the middle of a function, it can be moved to the start of the function, to save some cpu cycles in cases where the tracing controller is NULL?

Reply:
@gireeshpunathil Yes, but the lines between the start of the function and here take ownership of memory, so moving this line would cause a memory leak?

Reply:
do you mean the `std::unique_ptr<v8::ConvertableToTraceFormat>` objects? cant those be allocated only in the normal path? i.e., if `TracingController` object is not null?

Reply:
@gireeshpunathil They are not being allocated here, though – this function only takes ownership of them.